components/dashboard/src/start/VerifyModal.tsx

```diff
@@ -0,0 +1,220 @@
+                    </Alert>
+                ) : (
+                    <></>
+                )}
+                <div className="mt-4">
+                    <h4>Mobile Phone Number</h4>
+                    {/* HACK: Below we are adding a dummy dom element that is not visible, to reference the classes so they are not removed by purgeCSS. */}
+                    <input type="tel" className="hidden intl-tel-input country-list" />
+                    <PhoneInput
+                        autoFocus={true}
+                        containerClassName={"allow-dropdown w-full intl-tel-input"}
+                        inputClassName={"w-full"}
+                        allowDropdown={true}
+                        defaultCountry={""}
+                        autoHideDialCode={false}
+                        onPhoneNumberChange={(isValid, phoneNumberRaw, countryData) => {
+                            let phoneNumber = phoneNumberRaw;
+                            if (!phoneNumber.startsWith("+") && !phoneNumber.startsWith("00")) {
+                                phoneNumber = "+" + countryData.dialCode + phoneNumber;
+                            }
+                            setState({
+                                ...state,
+                                phoneNumber,
+                                phoneNumberValid: isValid,
+                            });
+                        }}
+                    />
+                </div>
+            </Modal>
+        );
+    } else if (!state.verified) {
+        const isTokenFilled = () => {
+            return state.token && /\d{6}/.test(state.token);
+        };
+        const verifyToken = async () => {
+            const verified = await getGitpodService().server.verifyPhoneNumberVerificationToken(
+                state.phoneNumber!,
+                state.token!,
+            );
+            if (verified) {
+                setState({
+                    ...state,
+                    verified: true,
+                    message: undefined,
+                });
+            } else {
+                setState({
+                    ...state,
+                    message: {
+                        type: "error",
+                        text: `Invalid verification code.`,
+                    },
+                });
+            }
+            return verified;
+        };
+
+        const reset = () => {
+            setState({
+                ...state,
+                sent: undefined,
+                message: undefined,
+                token: undefined,
+            });
+        };
+        return (
+            <Modal
+                onClose={() => {}}
+                closeable={false}
+                onEnter={verifyToken}
+                title="User Validation Required"
+                buttons={
+                    <div>
+                        <button className="ml-2" disabled={!isTokenFilled()} onClick={verifyToken}>
+                            Validate Account
```

Comment:
**suggestion**: I noticed we've skipped the _verification in progress_ step from the initial design specs in https://github.com/gitpod-io/gitpod/issues/11339#issuecomment-1183387915 which is fine, but what do you think of injecting a loading state inside the button here with the spinner icon? FWIW, we've used this pattern for running prebuilds in the past, see relevant diff below.

**thought**: We could use the same pattern also for sending the SMS via code since we rely on an external service for this action.

**thought**: The loading state of the button is another great example use case where a **button component** could be used. 💭 

https://github.com/gitpod-io/gitpod/blob/31332971069a926428151aae2eacac947ed2db21/components/dashboard/src/projects/Prebuild.tsx#L160-L169

Re-posting the verification in progress design from https://github.com/gitpod-io/gitpod/issues/11339#issuecomment-1183387915 for visibility:

> | Verification (Progress) |
> |-|
> | <img width="1440" alt="UserValidationModalVerificationInProgress" src="https://user-images.githubusercontent.com/120486/178774627-8e8b8c07-bcc1-4b87-b461-3a229f1071d3.png"> |